sensor/extrinsic_reflector_based_calibrator/src/extrinsic_reflector_based_calibrator.cpp

```diff
@@ -1280,21 +1336,169 @@ void ExtrinsicReflectorBasedCalibrator::calibrateSensors()
+    std::vector<double> crossval_calibrated_2d_yaw_error_vector;
+    total_crossval_calibrated_2d_distance_error = 0.0;
+    total_crossval_calibrated_2d_yaw_error = 0.0;
+
+    findCombinations(tracks_size - 1, num_of_samples, curr, 0, combinations);
+
+    RCLCPP_INFO(
+      this->get_logger(), "The number of combinations is: %f, tracks_size: %d, num_of_samples: %d",
+      static_cast<double>(combinations.size()), tracks_size, num_of_samples);
+
+    // random select the combinations if the number of combinations is too large
+    if (combinations.size() > static_cast<std::size_t>(parameters_.max_number_of_combinations)) {
+      std::random_device rd;
+      std::mt19937 mt(rd());
+      std::shuffle(combinations.begin(), combinations.end(), mt);
+      combinations.resize(parameters_.max_number_of_combinations);
+      RCLCPP_INFO(
+        this->get_logger(), "The number of new combinations size is : %f",
+        static_cast<double>(combinations.size()));
+    }
+
+    for (const auto & combination : combinations) {
+      // clear the lidar radar pcs
+      crossval_lidar_points_pcs->clear();
+      crossval_radar_points_rcs->clear();
+      // calculate the transformation.
+      for (int j = 0; j < num_of_samples; j++) {
+        crossval_lidar_points_pcs->emplace_back(lidar_points_pcs->points[combination[j]]);
+        crossval_radar_points_rcs->emplace_back(radar_points_rcs->points[combination[j]]);
+      }
+      crossval_estimator.estimateRigidTransformation(
+        *crossval_lidar_points_pcs, *crossval_radar_points_rcs,
+        crossval_radar_to_parent_transformation);
+      crossval_calibrated_2d_radar_to_parent_transformation =
+        crossval_radar_to_parent_transformation.cast<double>();
+      crossval_calibrated_2d_radar_to_parent_transformation.translation().z() =
+        (initial_radar_to_lidar_eigen_ * parent_to_lidar_eigen_.inverse()).translation().z();
+      crossval_calibrated_2d_radar_to_lidar_transformation =
+        crossval_calibrated_2d_radar_to_parent_transformation * parent_to_lidar_eigen_;
+
+      // calculate the error.
+      auto [crossval_calibrated_2d_distance_error, crossval_calibrated_2d_yaw_error] =
+        computeCalibrationError(crossval_calibrated_2d_radar_to_lidar_transformation);
+
+      total_crossval_calibrated_2d_distance_error += crossval_calibrated_2d_distance_error;
+      total_crossval_calibrated_2d_yaw_error += crossval_calibrated_2d_yaw_error;
+      crossval_calibrated_2d_distance_error_vector.push_back(crossval_calibrated_2d_distance_error);
+      crossval_calibrated_2d_yaw_error_vector.push_back(crossval_calibrated_2d_yaw_error);
+    }
+
+    auto calculate_std = [](std::vector<double> & data, double mean) -> double {
+      double sum = 0.0;
+      for (size_t i = 0; i < data.size(); i++) {
+        sum += (data[i] - mean) * (data[i] - mean);
+      }
+      double variance = sum / data.size();
+      return sqrt(variance);
```

Comment:
Can you use std::sqrt please? (std ones are templated)

Reply:
Sure, fix with the new commit https://github.com/tier4/CalibrationTools/pull/145/commits/55f555fdbbeece094a52aaabf448e4baa6f9fc78